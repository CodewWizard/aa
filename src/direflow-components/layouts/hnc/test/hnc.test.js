import React from "react";
import { create } from "react-test-renderer";
import HeaderNavigationContent from "../hnc";

describe("HeaderNavigationContent", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      store: {
        getState: jest.fn().mockReturnValue({
          loggedInUserReducer: {
            encodedJWT: "abc123",
            decodedJWT: {
              FirstName: "John",
              LastName: "Doe",
              role: "Admin",
            },
          },
          applicationReducer: {},
        }),
      },
      sharedSetNavigation: jest.fn(),
      onMenuSelected: jest.fn(),
      Menus: [],
      userLogout: jest.fn(),
      checkValidUser: jest.fn(),
    };
    wrapper = create(<HeaderNavigationContent {...props} />);
  });

  afterEach(() => {
    props = null;
    wrapper = null;
  });

  it("renders correctly", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
