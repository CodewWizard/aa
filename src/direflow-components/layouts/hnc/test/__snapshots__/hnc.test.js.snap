// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavigationContent renders correctly 1`] = `
<div
  className="wrapper"
>
  <nav
    className="border-end bgWhite commonShadow active"
    id="sidebar"
  >
    <ul
      className="list-unstyled components"
    />
  </nav>
  <div
    className="top-navbar bgWhite border-bottom commonShadow"
  >
    <nav
      className="navbar navbar-expand-lg h-100 py-0"
    >
      <div
        className="container-fluid h-100 px-0"
      >
        <a
          className="navbar-brand d-flex align-items-center ps-3 nohandcursor"
        >
          <img
            alt="Cogitate"
            className="headerLogo"
            src="images/logo.png"
            width="100px"
          />
          <span
            className="ms-3 ps-3 position-relative py-2"
            id="sidebarCollapse"
          />
        </a>
        <div
          className="collapse navbar-collapse d-lg-block d-xl-block d-md-none d-none flex-grow-0 h-100"
          id="navbarSupportedContent"
        >
          <ul
            className="nav navbar-nav notifiContainer ml-auto h-100 commonShadow"
          >
            <li
              className="dropdown notifiBox d-flex align-items-center border-start nav-item h-100 active"
            >
              <ul
                className="dropdown-menu dropdown-menu-align boxAlign"
              >
                <h3
                  className="notification-list-name d-flex justify-content-between align-items-lg-center mb-0"
                >
                  Notifications
                  <span
                    className="clear-3"
                  >
                    Clear All
                  </span>
                </h3>
                <li
                  className="list__item"
                >
                  <a
                    className="px-0"
                  >
                    <div
                      className="row m-auto"
                    >
                      <div
                        className="col-md-2 d-flex"
                      >
                        <p
                          className="user-image"
                        >
                          IB
                        </p>
                      </div>
                      <div
                        className="col-md-8 px-0"
                      >
                        <p
                          className="messages"
                        >
                          <b>
                             Underwriter name 
                          </b>
                           has approved Application XXXX and Policy number generated is: SGIH00000000.
                        </p>
                        <p
                          className="rowDate"
                        >
                          May 05, 2021
                        </p>
                      </div>
                      <div
                        className="col-md-2 d-flex m-auto justify-content-center"
                      >
                        <i
                          aria-hidden="true"
                          className="fa fa-minus-circle rowDelete"
                        />
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </li>
            <li
              className="nav-item d-flex align-items-center border-start h-100"
            >
              <a
                className="nav-link userPanLink d-flex nohandcursor"
              >
                <div
                  className="userCode"
                >
                  <h6
                    className="m-auto"
                  >
                    J
                    D
                  </h6>
                </div>
                <div
                  className="d-flex flex-column userName"
                >
                  <h6
                    className="text-body mb-0 font-14"
                  >
                    JOHN
                     
                    DOE
                  </h6>
                  <span
                    className="text-black-50"
                  >
                    Underwriter
                  </span>
                </div>
              </a>
            </li>
            <li
              className="nav-item d-flex align-items-center border-start h-100"
            >
              <a
                className="nav-link d-grid"
                id="logout"
                onClick={[Function]}
              >
                <img
                  alt=""
                  className="m-auto"
                  src="images/logout-ico.png"
                />
                <span
                  className="primaryColor1"
                >
                  Logout
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="active"
    id="content"
  />
</div>
`;
